### DISCUSSION TITLE
How do I get a random row

### DISCUSSION THREAD

**Discussion by ShayBox** [SeaQL/sea-orm#1343] [Q&A]
I'm currently fetching all rows and picking a random one manually, but this has grown to become very time consuming with 100k rows, how would I fetch a random row via a query?
```rust
let servers = Server::find()
    .all(db.0)
    .await
    .map_err(InternalServerError)?;

let Some(server) = servers.choose(&mut rand::thread_rng()) else {
    return Err(poem::Error::from_status(StatusCode::NO_CONTENT))
};
```

**Comment by tyt2y3** [comment]
If you can ensure the table's primary key is contiguous (otherwise use a sequence column), you can just generate a random number between (0, MAX_PK).

**Reply by ShayBox** [reply]
That's a pretty good solution, I hadn't thought about that

I was trying to figure out how to use the `Random` function with `ORDER BY` but I can't figure it out  
https://github.com/SeaQL/sea-query/blob/master/src/func.rs#L24

I was able to get it working with raw sql from string
```rust
let Some(server) = Server::find()
    .from_raw_sql(Statement::from_string(
        DbBackend::MySql,
        r#"SELECT `Server`.`address`, `Server`.`host`, `Server`.`port`, `Server`.`version`, `Server`.`protocol`, `Server`.`max_players`, `Server`.`online_players`, `Server`.`response` FROM `Server` ORDER BY RAND() LIMIT 1"#.to_owned(),
    ))
    .one(db.0)
    .await
    .map_err(InternalServerError)?
else {
    return Err(poem::Error::from_status(StatusCode::NO_CONTENT))
};
```
but I can't figure out the code needed for SeaQL ORM to generate that query, I just added `ORDER BY RAND() LIMIT 1` to the end of the generated query string from that same code without `.from_raw_sql`, and even with `.one` adding `LIMIT 1` seemed to speed it up dramatically
